apisix/init.lua

```diff
@@ -192,10 +211,16 @@ function _M.http_ssl_phase()
         if err then
             core.log.error("failed to fetch ssl config: ", err)
         end
+        core.log.error("failed to find any SSL certificate by SNI: ", sni)
         ngx_exit(-1)
     end
-end
 
+    ok ,err = apisix_ssl.set_protocols_by_clienthello(ngx_ctx.matched_ssl.value.ssl_protocols)
```

Comment:
```suggestion
    ok, err = apisix_ssl.set_protocols_by_clienthello(ngx_ctx.matched_ssl.value.ssl_protocols)
```

Reply:
BTW, I'm not sure if I should use `ngx_ctx` or `api_ctx` as a params.

Reply:
`api_ctx` has been released and should not be used.
Can only be selected in `ngx_ctx` and `local variables`.